### PR TITLE
[RISCV][GISel] Lower G_FCONSTANT to constant pool load without F or D.

### PR DESCRIPTION
**Pull request by topperc**
I used an IR test because it was easier than constructing different MIR test for each type of addressing.

### CODE REVIEW

File: llvm/test/CodeGen/RISCV/GlobalISel/constantpool.ll [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+; RV64-MEDIUM-NEXT:    ret
+;
+; RV64-PIC-LABEL: constpool_f32:
+; RV64-PIC:       # %bb.0:
+; RV64-PIC-NEXT:  .Lpcrel_hi0:
+; RV64-PIC-NEXT:    auipc a1, %pcrel_hi(.LCPI0_0)
+; RV64-PIC-NEXT:    ld a1, %pcrel_lo(.Lpcrel_hi0)(a1)
+; RV64-PIC-NEXT:    sd a1, 0(a0)
+; RV64-PIC-NEXT:    ret
+  store double 1.0, ptr %p
```
Comment (tschuett) [inline llvm/test/CodeGen/RISCV/GlobalISel/constantpool.ll:76]: `float` ?

Comment (topperc) [reply llvm/test/CodeGen/RISCV/GlobalISel/constantpool.ll:76]: thanks

File: llvm/lib/Target/RISCV/GISel/RISCVInstructionSelector.cpp [hunk 2]
```diff
@@ -932,6 +936,98 @@ bool RISCVInstructionSelector::selectJumpTable(MachineInstr &MI,
   return false;
 }
 
+// FIXME: This is very similar to selectGlobalValue. Merge somehow?
```
Comment (michaelmaitland) [inline llvm/lib/Target/RISCV/GISel/RISCVInstructionSelector.cpp:939]: Should we stack this change on https://github.com/llvm/llvm-project/pull/72769?

File: llvm/lib/Target/RISCV/GISel/RISCVLegalizerInfo.cpp [hunk 3]
```diff
@@ -273,7 +274,9 @@ RISCVLegalizerInfo::RISCVLegalizerInfo(const RISCVSubtarget &ST)
   getActionDefinitionsBuilder(G_IS_FPCLASS)
       .customIf(all(typeIs(0, s1), typeIsScalarFPArith(1, ST)));
 
-  getActionDefinitionsBuilder(G_FCONSTANT).legalIf(typeIsScalarFPArith(0, ST));
+  getActionDefinitionsBuilder(G_FCONSTANT)
+      .legalIf(typeIsScalarFPArith(0, ST))
+      .lowerFor({s32, s64});
```
Comment (michaelmaitland) [inline llvm/lib/Target/RISCV/GISel/RISCVLegalizerInfo.cpp:279]: How do we know that the `GFConstantMI.getOperand(1).getFPImm()` is in the constant pool?

**Review by michaelmaitland — APPROVED** [review]
LGTM.

**Pull request merged**